packages/server/src/channels/telegram/channel.ts

```diff
@@ -25,16 +26,26 @@ export class TelegramChannel extends Channel<TelegramConduit> {
 
   async setupRoutes() {
     this.router.use(
-      '/',
+      '/:token',
       this.asyncMiddleware(async (req, res) => {
         // This is done to make forwarding work
-        req.url = '/'
+        req.url = `/${req.params.token}`
 
         const conduit = res.locals.conduit as TelegramConduit
-        conduit.callback(req, res)
+        if (this.validate(conduit, req)) {
+          conduit.callback(req, res)
+        } else {
+          this.logger.error("Request validation failed. Request probably didn't come from Telegram.")
+
+          res.status(401).send('Auth token invalid')
+        }
       })
     )
 
     this.printWebhook()
   }
+
+  private validate(conduit: TelegramConduit, req: Request): boolean {
+    return req.params.token === conduit.config.botToken
+  }
```

Comment:
It's kind of an unnecessary overhead to extract this as a function here. There original function is already tiny so putting that in there won't hurt readability.

Reply:
I totally agree, that in this case, the function is unnecessary, but I think it would be nice if all channels implemented a validate function so we can maybe factorize the code a little and, by default, add this function as middleware or something like that!